src/DynamoCore/Graph/Workspaces/HomeWorkspaceModel.cs

```diff
@@ -433,6 +433,18 @@ private void OnUpdateGraphCompleted(AsyncTask task)
             var updateTask = (UpdateGraphAsyncTask)task;
             var messages = new Dictionary<Guid, string>();
 
+            // Mark all modified nodes as being updated.
+            // 
+            // In addition to marking modified nodes as being updated, their 
+            // warning states are cleared (which include the tool-tip).
+            foreach (var modifiedNode in updateTask.ModifiedNodes)
```

Comment:
Come to think about it... `WasInvolvedInExecution` is either referenced by `UpdateGraphAsyncTask` or `RequestValueUpdateAsyncTask` classes. If we move `WasInvolvedInExecution` to UI thread, then there will still be race condition. I'm thinking the solution might just be moving `WasInvolvedInExecution` reads/writes to inside `AsyncTask` derived classes (the above two). The only complication left after that would then be in `helixwatch3dnodeviewmodel.cs` when `UpdateUpstream` is triggered. Maybe we can also turn that into another `Action` for `DelegateBasedAsyncTask`?